src/Driver.cpp

```diff
@@ -28,10 +30,15 @@ void Driver::open(std::string const& uri)
     openURI(uri);
 }
 
-void Driver::read()
+bool Driver::read()
 {
-    readPacket(&buffer[0], buffer.size());
-    parseMessage(&buffer[4],buffer.size()-4);
+    int msg_size = readPacket(&buffer[0], buffer.size());
+    if(msg_size == KVH_1750_MSG_SIZE)
```

Comment:
> It basically shows that you don't understand what contracts you need to have enforced in your codebase, a.k.a. that you don't understand the codebase ("not understanding" is fine, defending your lack of understanding is not). After a few hundred of "no harm in testing X" you basically make everyone's life a living hell - including yours a few months from now. As a direct example, here, you return a boolean on a method that can only return true. And then test for it in a component, creating two dead paths.

It wasn't my intention to create any dead paths. What we have here is a method blindly parsing the given buffer and a method filling the buffer and returning it's size. If we want to enforce any contracts in such cases, there should be at least a comment explaining it, otherwise everyone who looks into this code has to check if this is not a possible point of failure.

> Moreover, it shows that the analysis you did for a bug (with the commit messages I've seen, I am assuming that you're trying to fix a bug) is wrong, since the "fix" most probably does not change anything.

The bug is that extractPacket returns the whole buffer, what I describe in the first comment of this PR.

> If you're afraid of having a bug in extractPacket, thoroughly unit test it.

Sure. I can add a unit test.